cmd/lp-colouring/colouring.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"sync/atomic"
+
+	"github.com/ScottSallinen/lollipop/graph"
+	"github.com/kelindar/bitmap"
+)
+
+const EmptyColour = math.MaxUint32
+
+type VertexProperty struct {
+	NbrColours sync.Map
+	WaitCount  int64
+	Colour     uint32
+}
+
+func (p *VertexProperty) String() string {
+	s := fmt.Sprintf("{%d,%d,[", p.Colour, p.WaitCount)
+	p.NbrColours.Range(func(key, value any) bool {
+		s += fmt.Sprintf("%d:%d,", key, value)
+		return true
+	})
+	return s + "]}"
+}
+
+func hash(id uint32) (hash uint32) {
+	// TODO: dummy hash function
+	return id
+}
+
+func comparePriority(p1, p2 uint32, id1, id2 uint32) bool {
```

Comment:
I suggest this function only take indexes as parameters, as indices are already unique. Ideally the rest of the code does not need to call the hash func.

i.e.,

hp1 = hash(p1)
hp2 = hash(p2)
return  hp1 > hp1  || (hp1 == hp2 && p1> p2)